services:
  db:
    image: postgres
    environment:
      - POSTGRES_PASSWORD=postgres
    ports:
      - "5432:5432"
    networks:
      artconomy:
        ipv4_address: 10.5.0.2
  nginx:
    hostname: 'artconomy.vulpinity.com'
    build:
      dockerfile: docker/nginx_Dockerfile
      context: .
    container_name: ng01
    ports:
      - "443:443"
    volumes:
      - ./config/nginx/nginx.conf:/etc/nginx/nginx.conf
      - ./config/nginx/conf.d:/etc/nginx/conf.d
      - ./dev_keys:/etc/nginx/dev_keys
      - ./:/var/www
    networks:
      artconomy:
        ipv4_address: 10.5.0.3
    depends_on:
      - web
      - frontend
      - companion
  redis:
    image: redis
    ports:
      - "6379:6379"
    networks:
      artconomy:
        ipv4_address: 10.5.0.4
  web:
    ports:
      - "8002:8002"
    networks:
      artconomy:
        ipv4_address: 10.5.0.5
    extends:
      file: common-services.yml
      service: django

  tgbot:
    extends:
      file: common-services.yml
      service: django
    networks:
      artconomy:
        ipv4_address: 10.5.0.6
    command: python manage.py tg_bot

  frontend:
    command: npm --prefix /app/ run serve
    image: node:20.17-alpine
    networks:
      artconomy:
        ipv4_address: 10.5.0.7
    ports:
      - "8001:8001"
    volumes:
      - ./:/app/

  companion:
    command: npx --prefix /app/ companion
    environment:
      - NODE_TLS_REJECT_UNAUTHORIZED=0
      - COMPANION_SECRET=aseriuehw4giluwesrbilunsdfijke4twesdf
      - COMPANION_DOMAIN=artconomy.vulpinity.com
      - COMPANION_PROTOCOL=https
      - COMPANION_CLIENT_ORIGINS=true
      - COMPANION_DATADIR=/app/media/companion/
      - COMPANION_PATH=/companion/
      - COMPANION_PORT=3020
    image: node:20.17-alpine
    networks:
      artconomy:
        ipv4_address: 10.5.0.8
    ports:
      - "3020:3020"
    volumes:
      - ./:/app/


networks:
  artconomy:
    driver: bridge
    ipam:
      driver: default
      config:
        - subnet: 10.5.0.0/16
          gateway: 10.5.0.1
